### PR TITLE
some fixes for container deployment

### PR DESCRIPTION
**Pull request by rootfs**
in containerized deployment, use the right interface name variable in mon; remove one line inline files

cleanup on docker osd tasks

@alfredodeza @leseb

### CODE REVIEW

File: roles/ceph-osd/tasks/docker/start_docker_osd.yml [hunk 1]
```diff
@@ -15,25 +15,31 @@
...
-    image: "{{ ceph_osd_docker_username }}/{{ ceph_osd_docker_imagename }}"
-    name: "{{ ansible_hostname }}-osd-prepare-{{ item.0 | regex_replace('/', '') }}"
-    net: host
-    pid: host
-    state: started
-    privileged: yes
-    env: "OSD_DEVICE={{ item.0 }},{{ ceph_osd_docker_prepare_env }}"
-    volumes: "/var/lib/ceph:/var/lib/ceph,/etc/ceph:/etc/ceph,/dev:/dev,/run:/run"
+  shell: |   
+    docker run -ti --net=host \
```
Comment (leseb) [inline roles/ceph-osd/tasks/docker/start_docker_osd.yml]: `run -ti`?
why not `run -d`?

Comment (rootfs) [inline roles/ceph-osd/tasks/docker/start_docker_osd.yml]: the problem I see is that docker module doesn't watch container exit status, so the task may return before the command is finished (same with `docker -d`). Then when next task (osd activate) is started, the osd is still being prepared and that causes problem.

File: group_vars/all.docker.sample [hunk 2]
```diff
@@ -27,7 +27,7 @@ dummy:
 #ceph_osd_docker_username: ceph
 #ceph_osd_docker_imagename: daemon
 #ceph_osd_docker_extra_env: "CEPH_DAEMON=OSD_CEPH_DISK_ACTIVATE" # comma separated variables
-#ceph_osd_docker_prepare_env: "CEPH_DAEMON=OSD_CEPH_DISK_PREPARE,OSD_FORCE_ZAP=1" # comma separated variables
+#ceph_osd_docker_prepare_env: "OSD_FORCE_ZAP=1" # comma separated variables
```
Comment (leseb) [inline group_vars/all.docker.sample]: can you also edit `roles/ceph-osd/defaults/main.yml` and run `generate_group_vars_sample.sh`?

File: roles/ceph-osd/tasks/docker/start_docker_osd.yml (hunk 1 above)
Comment (leseb) [inline roles/ceph-osd/tasks/docker/start_docker_osd.yml]: Ok I understand. Makes sense.

File: group_vars/all.docker.sample (hunk 2 above)
Comment (rootfs) [inline group_vars/all.docker.sample]: done, thanks!

**Comment by rootfs** [comment]
tested containerized deployment on centos and RHEL Atomic.
@leseb @alfredodeza PTAL

File: group_vars/osds.sample [hunk 3]
```diff
@@ -131,7 +131,7 @@ dummy:
 #osd_containerized_deployment_with_kv: false
 #kv_type: etcd
 #kv_endpoint: 127.0.0.1
-#ceph_osd_docker_prepare_env: ""
+#ceph_osd_docker_prepare_env: "CEPH_DAEMON=OSD_CEPH_DISK_PREPARE"
```
Comment (leseb) [inline group_vars/osds.sample]: why do you have a different content now? 
both  `group_vars/all.docker.sample`  and `group_vars/osds.sample` should share the same content.

Comment (rootfs) [inline group_vars/osds.sample]: re-generated

**Comment by rootfs** [comment]
@leseb  @dang PTAL thanks

**Comment by leseb** [comment]
test this please

**Pull request merged**